Testing/Unit/PowerShell/Connection/Connect-Tenant.Tests.ps1

```diff
@@ -0,0 +1,235 @@
+Import-Module (Join-Path -Path $PSScriptRoot -ChildPath "../../../../PowerShell/ScubaGear/Modules/Connection/Connection.psm1") -Function 'Connect-Tenant' -Force
+
+InModuleScope Connection {
+    Describe -Tag 'Connection' -Name 'Connect-Tenant' {
+        BeforeAll {
+            function Connect-MgGraph {}
+            Mock -ModuleName Connection Connect-MgGraph -MockWith {}
+            function Connect-PnPOnline {}
+            Mock -ModuleName Connection Connect-PnPOnline -MockWith {}
+            function Connect-SPOService {}
+            Mock -ModuleName Connection Connect-SPOService -MockWith {}
+            function Connect-MicrosoftTeams {}
+            Mock -ModuleName Connection Connect-MicrosoftTeams -MockWith {}
+            function Add-PowerAppsAccount {}
+            Mock -ModuleName Connection Add-PowerAppsAccount -MockWith {}
+            function Connect-EXOHelper {}
+            Mock -ModuleName Connection Connect-EXOHelper -MockWith {}
+            function Select-MgProfile {}
+            Mock -ModuleName Connection Select-MgProfile -MockWith {}
+            function Get-MgProfile {}
+            Mock -ModuleName Connection Get-MgProfile -MockWith {
+                [pscustomobject]@{
+                    Name = "alpha";
+                }
+            }
+            function Get-MgOrganization {}
+            Mock -ModuleName Connection Get-MgOrganization -MockWith {
+                return [pscustomobject]@{
+                    DisplayName     = "DisplayName";
+                    Name            = "DomainName";
+                    Id              = "TenantId";
+                    VerifiedDomains = @(
+                        @{
+                            isInitial = $false;
+                            Name      = "example.onmicrosoft.com"
+                        },
+                        @{
+                            isInitial = $true;
+                            Name      = "contoso.onmicrosoft.com"
+                        }
+                    )
+                }
+            }
+            Mock -CommandName Write-Progress {
+            }
+        }
+        Context "When interactively connecting to Commercial Endpoints" {
+            BeforeAll {
+                [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSUseDeclaredVarsMoreThanAssignments', 'M365Environment')]
+                $M365Environment = "commercial"
+            }
+            It 'With -ProductNames "aad", connects to Microsoft Graph' {
+                $ProductNames = @("aad")
+                $FailedAuthList = Connect-Tenant -ProductNames $ProductNames -M365Environment $M365Environment
+                $FailedAuthList.Length | Should -Be 0
```

Comment:
Consider add check if proper connection authorization (i.e., Connecet-MgGraph) is called.  Can also be extended to insure called with correct parameters,

```suggestion
                $FailedAuthList.Length | Should -Be 0
                Should -Invoke -CommandName Connect-MgGraph -Times 1
```